src/onboarding/registration/NameAndPicture.tsx

```diff
@@ -73,17 +98,22 @@ function NameAndPicture({ navigation, route }: Props) {
         )
       },
       headerRight: () =>
-        skipUsername && (
+        showSkipButton && (
           <TopBarTextButton
             title={t('skip')}
-            onPress={onPressSkip}
+            onPress={goToNextScreen}
             titleStyle={{ color: colors.goldDark }}
           />
         ),
     })
   }, [navigation, choseToRestoreAccount, step, totalSteps, nameInput])
 
   const goToNextScreen = () => {
+    try {
```

Comment:
Looking through the statsig sdk https://github.com/statsig-io/node-js-server-sdk it appears that the only time this throws is when statsig is uninitialized. Can we guarantee that its initialized by this point in the code & not need the try/catch?

Reply:
If not, this also seems like it could use a helper function so we don't need to wrap every single metric log in a try/catch

Reply:
I think we'd be getting errors [client sdk](https://github.com/statsig-io/js-client) but your point still stands. I think we can guarantee initialize is called by that point, but they may be offline (e.g. on airplane mode) so Statsig won't connect. Also I didn't realize their code was open source, so I had assumed that this failing to connect could lead to other errors. Looking at it now I don't think that's the case, so I can remove the try/catch

Reply:
Removing the try-catch leads to e2e errors, so I've added it back in. I also forgot that I wrote an [issue](https://linear.app/valora/issue/ACT-467/segmentstatsig-integration) to log events via Segment rather than Statsig, so I think this is just a temporary setup anyhow.